// @daada/shared/src/schemas/index.ts
import { z } from "zod";

// ── Auth ──────────────────────────────────────────────────────────────────────

export const phoneSchema = z
  .string()
  .regex(/^\+?[1-9]\d{8,14}$/, "Numéro de téléphone invalide");

export const loginSchema = z.object({
  phone: phoneSchema,
});

export const verifyOtpSchema = z.object({
  phone: phoneSchema,
  code: z.string().length(6, "Le code doit contenir 6 chiffres"),
});

// ── Order ─────────────────────────────────────────────────────────────────────

export const orderItemSchema = z.object({
  menuItemId: z.string().uuid(),
  quantity: z.number().int().positive().max(20),
  notes: z.string().max(200).optional(),
});

export const createOrderSchema = z.object({
  restaurantId: z.string().uuid(),
  items: z.array(orderItemSchema).min(1),
  deliveryAddress: z.string().min(5).max(500),
  deliveryLatitude: z.number(),
  deliveryLongitude: z.number(),
  paymentMethod: z.enum(["orange_money", "mtn_momo", "cash_on_delivery"]),
  notes: z.string().max(500).optional(),
});

// ── User ──────────────────────────────────────────────────────────────────────

export const updateProfileSchema = z.object({
  name: z.string().min(2).max(100).optional(),
  email: z.string().email().optional(),
});
