// @daada/ui/src/components/Badge.tsx
import * as React from "react";
import { cn } from "../utils/cn";

type BadgeVariant = "default" | "primary" | "success" | "warning" | "danger" | "premium" | "free";

interface BadgeProps {
  variant?: BadgeVariant;
  children: React.ReactNode;
  className?: string;
}

const variantStyles: Record<BadgeVariant, string> = {
  default: "bg-gray-100 text-gray-700 dark:bg-gray-800 dark:text-gray-300",
  primary: "bg-primary-100 text-primary-700 dark:bg-primary-900/30 dark:text-primary-300",
  success: "bg-green-100 text-green-700 dark:bg-green-900/30 dark:text-green-300",
  warning: "bg-amber-100 text-amber-700 dark:bg-amber-900/30 dark:text-amber-300",
  danger: "bg-red-100 text-red-700 dark:bg-red-900/30 dark:text-red-300",
  premium: "bg-gradient-to-r from-amber-500 to-yellow-400 text-white shadow-sm",
  free: "bg-green-500 text-white",
};

export function Badge({ variant = "default", children, className }: BadgeProps) {
  return (
    <span
      className={cn(
        "inline-flex items-center rounded-full px-2.5 py-0.5 text-xs font-semibold",
        variantStyles[variant],
        className,
      )}
    >
      {children}
    </span>
  );
}

export function Tag({ children, className }: { children: React.ReactNode; className?: string }) {
  return (
    <span
      className={cn(
        "inline-flex items-center rounded-lg border border-gray-200 bg-white px-2 py-1 text-xs",
        "text-gray-600 dark:border-gray-700 dark:bg-gray-800 dark:text-gray-400",
        className,
      )}
    >
      {children}
    </span>
  );
}
