// @daada/ui/src/components/CategoryTab.tsx
import * as React from "react";
import { cn } from "../utils/cn";

interface Category {
  id: string;
  label: string;
  icon: string;
}

interface CategoryTabProps {
  categories: Category[];
  selected: string;
  onChange: (id: string) => void;
  className?: string;
}

export function CategoryTab({ categories, selected, onChange, className }: CategoryTabProps) {
  return (
    <div
      className={cn(
        "flex gap-2 overflow-x-auto pb-2 scrollbar-none",
        "[-ms-overflow-style:none] [scrollbar-width:none] [&::-webkit-scrollbar]:hidden",
        className,
      )}
      role="tablist"
    >
      {categories.map((cat) => (
        <button
          key={cat.id}
          role="tab"
          aria-selected={selected === cat.id}
          onClick={() => onChange(cat.id)}
          className={cn(
            "flex shrink-0 items-center gap-2 rounded-full px-4 py-2 text-sm font-medium",
            "transition-all duration-150 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-primary",
            selected === cat.id
              ? "bg-primary text-white shadow-sm"
              : "bg-gray-100 text-gray-600 hover:bg-gray-200 dark:bg-gray-800 dark:text-gray-400 dark:hover:bg-gray-700",
          )}
        >
          <span aria-hidden="true">{cat.icon}</span>
          <span>{cat.label}</span>
        </button>
      ))}
    </div>
  );
}
