// @daada/ui/src/components/Input.tsx
import * as React from "react";
import { cn } from "../utils/cn";

interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {
  label?: string;
  error?: string;
  leftElement?: React.ReactNode;
  rightElement?: React.ReactNode;
}

export const Input = React.forwardRef<HTMLInputElement, InputProps>(
  ({ label, error, leftElement, rightElement, className, id, ...props }, ref) => {
    const inputId = id ?? label?.toLowerCase().replace(/\s+/g, "-");
    return (
      <div className="flex flex-col gap-1.5">
        {label && (
          <label htmlFor={inputId} className="text-sm font-medium text-gray-700 dark:text-gray-300">
            {label}
          </label>
        )}
        <div className="relative flex items-center">
          {leftElement && (
            <span className="absolute left-3 text-gray-400">{leftElement}</span>
          )}
          <input
            ref={ref}
            id={inputId}
            className={cn(
              "w-full rounded-xl border border-gray-200 bg-white px-4 py-3 text-sm",
              "placeholder:text-gray-400 dark:border-gray-700 dark:bg-gray-900 dark:text-white",
              "focus:border-primary focus:outline-none focus:ring-2 focus:ring-primary/20",
              "transition-colors duration-150",
              "disabled:cursor-not-allowed disabled:opacity-50",
              error && "border-red-400 focus:border-red-500 focus:ring-red-200",
              leftElement && "pl-10",
              rightElement && "pr-10",
              className,
            )}
            {...props}
          />
          {rightElement && (
            <span className="absolute right-3 text-gray-400">{rightElement}</span>
          )}
        </div>
        {error && <p className="text-xs text-red-500">{error}</p>}
      </div>
    );
  },
);
Input.displayName = "Input";

interface PhoneInputProps extends Omit<InputProps, "type" | "placeholder"> {
  countryCode?: string;
}

export function PhoneInput({ countryCode = "+237", className, ...props }: PhoneInputProps) {
  return (
    <div className="flex gap-2">
      <div className="flex h-12 items-center rounded-xl border border-gray-200 bg-white px-3 text-sm font-medium dark:border-gray-700 dark:bg-gray-900">
        <span className="mr-1">🇨🇲</span>
        <span className="text-gray-600 dark:text-gray-400">{countryCode}</span>
      </div>
      <Input
        type="tel"
        placeholder="6XX XXX XXX"
        className={cn("flex-1", className)}
        {...props}
      />
    </div>
  );
}

export function SearchInput({ className, ...props }: InputProps) {
  return (
    <Input
      type="search"
      leftElement={
        <svg className="h-4 w-4" fill="none" stroke="currentColor" viewBox="0 0 24 24">
          <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z" />
        </svg>
      }
      placeholder="Rechercher un plat..."
      className={cn("pl-10", className)}
      {...props}
    />
  );
}
