// @daada/ui/src/components/MenuItemCard.tsx
import * as React from "react";
import { cn } from "../utils/cn";
import { PriceDisplay } from "./PriceDisplay";
import { Badge } from "./Badge";

interface MenuItemCardProps {
  id: string;
  name: string;
  description?: string;
  price: number;
  imageUrl?: string;
  isPopular?: boolean;
  isAvailable?: boolean;
  preparationTime?: number;
  onAdd?: (id: string) => void;
  className?: string;
}

export function MenuItemCard({
  id,
  name,
  description,
  price,
  imageUrl,
  isPopular = false,
  isAvailable = true,
  preparationTime,
  onAdd,
  className,
}: MenuItemCardProps) {
  return (
    <div
      className={cn(
        "group overflow-hidden rounded-2xl border border-gray-100 bg-white transition-all duration-200",
        "hover:shadow-lg hover:-translate-y-0.5",
        "dark:border-gray-800 dark:bg-gray-900",
        !isAvailable && "opacity-60",
        className,
      )}
    >
      {/* Image */}
      <div className="relative h-48 overflow-hidden bg-gray-100 dark:bg-gray-800">
        {imageUrl ? (
          <img
            src={imageUrl}
            alt={name}
            className="h-full w-full object-cover transition-transform duration-300 group-hover:scale-105"
            loading="lazy"
          />
        ) : (
          <div className="flex h-full items-center justify-center text-5xl">🍽️</div>
        )}
        {isPopular && (
          <div className="absolute left-3 top-3">
            <Badge variant="primary">⭐ Populaire</Badge>
          </div>
        )}
        {!isAvailable && (
          <div className="absolute inset-0 flex items-center justify-center bg-black/40">
            <span className="rounded-lg bg-white/90 px-3 py-1 text-sm font-medium text-gray-700">
              Indisponible
            </span>
          </div>
        )}
      </div>

      {/* Content */}
      <div className="p-4">
        <h3 className="font-semibold text-gray-900 dark:text-white line-clamp-1">{name}</h3>
        {description && (
          <p className="mt-1 text-sm text-gray-500 dark:text-gray-400 line-clamp-2">{description}</p>
        )}
        {preparationTime && (
          <p className="mt-1 flex items-center gap-1 text-xs text-gray-400">
            <svg className="h-3 w-3" fill="none" stroke="currentColor" viewBox="0 0 24 24">
              <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z" />
            </svg>
            {preparationTime} min
          </p>
        )}
        <div className="mt-3 flex items-center justify-between">
          <PriceDisplay amount={price} size="md" />
          {isAvailable && onAdd && (
            <button
              onClick={() => onAdd(id)}
              aria-label={`Ajouter ${name} au panier`}
              className={cn(
                "flex h-9 w-9 items-center justify-center rounded-full",
                "bg-primary text-white transition-transform",
                "hover:scale-110 hover:bg-primary-600",
                "active:scale-95",
                "focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-primary",
              )}
            >
              <svg className="h-5 w-5" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2.5} d="M12 4v16m8-8H4" />
              </svg>
            </button>
          )}
        </div>
      </div>
    </div>
  );
}
