// @daada/ui/src/components/Navbar.tsx
import * as React from "react";
import { cn } from "../utils/cn";

interface NavbarProps {
  cartCount?: number;
  isLoggedIn?: boolean;
  onCartClick?: () => void;
  onLoginClick?: () => void;
  className?: string;
}

export function Navbar({
  cartCount = 0,
  isLoggedIn = false,
  onCartClick,
  onLoginClick,
  className,
}: NavbarProps) {
  const [menuOpen, setMenuOpen] = React.useState(false);

  return (
    <header
      className={cn(
        "sticky top-0 z-50 w-full border-b border-gray-100 bg-white/95 backdrop-blur",
        "dark:border-gray-800 dark:bg-background-dark/95",
        className,
      )}
    >
      <div className="mx-auto flex h-16 max-w-7xl items-center justify-between px-4 sm:px-6">
        {/* Logo */}
        <a href="/" className="flex items-center gap-2">
          <div className="flex h-9 w-9 items-center justify-center rounded-full bg-primary text-sm font-bold text-white shadow-sm">
            DF
          </div>
          <span className="font-display text-lg font-bold text-gray-900 dark:text-white">
            Daada<span className="text-primary"> Fast Food</span>
          </span>
        </a>

        {/* Desktop nav */}
        <nav className="hidden items-center gap-6 md:flex" aria-label="Navigation principale">
          {[
            { href: "/", label: "Accueil" },
            { href: "/menu", label: "Menu" },
            { href: "/cuisine", label: "Cuisine" },
            { href: "/about", label: "À propos" },
          ].map((link) => (
            <a
              key={link.href}
              href={link.href}
              className="text-sm font-medium text-gray-600 transition-colors hover:text-primary dark:text-gray-400 dark:hover:text-primary"
            >
              {link.label}
            </a>
          ))}
        </nav>

        {/* Actions */}
        <div className="flex items-center gap-3">
          {/* Cart */}
          <button
            onClick={onCartClick}
            aria-label={`Panier (${cartCount} articles)`}
            className="relative rounded-xl p-2 text-gray-600 hover:bg-gray-100 dark:text-gray-400 dark:hover:bg-gray-800 transition-colors"
          >
            <svg className="h-6 w-6" fill="none" stroke="currentColor" viewBox="0 0 24 24">
              <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M3 3h2l.4 2M7 13h10l4-8H5.4M7 13L5.4 5M7 13l-2.293 2.293c-.63.63-.184 1.707.707 1.707H17m0 0a2 2 0 100 4 2 2 0 000-4zm-8 2a2 2 0 11-4 0 2 2 0 014 0z" />
            </svg>
            {cartCount > 0 && (
              <span className="absolute -right-0.5 -top-0.5 flex h-5 w-5 items-center justify-center rounded-full bg-primary text-xs font-bold text-white">
                {cartCount > 9 ? "9+" : cartCount}
              </span>
            )}
          </button>

          {/* Auth */}
          {isLoggedIn ? (
            <a
              href="/profile"
              className="flex h-9 w-9 items-center justify-center rounded-full bg-primary text-sm font-bold text-white"
            >
              U
            </a>
          ) : (
            <button
              onClick={onLoginClick}
              className="hidden rounded-xl bg-primary px-4 py-2 text-sm font-semibold text-white hover:bg-primary-600 transition-colors md:block"
            >
              Connexion
            </button>
          )}

          {/* Mobile menu toggle */}
          <button
            onClick={() => setMenuOpen(!menuOpen)}
            className="rounded-xl p-2 text-gray-600 hover:bg-gray-100 dark:text-gray-400 dark:hover:bg-gray-800 md:hidden"
            aria-label="Menu"
          >
            {menuOpen ? (
              <svg className="h-6 w-6" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M6 18L18 6M6 6l12 12" />
              </svg>
            ) : (
              <svg className="h-6 w-6" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M4 6h16M4 12h16M4 18h16" />
              </svg>
            )}
          </button>
        </div>
      </div>

      {/* Mobile menu */}
      {menuOpen && (
        <div className="border-t border-gray-100 bg-white px-4 py-3 dark:border-gray-800 dark:bg-background-dark md:hidden">
          <nav className="flex flex-col gap-1">
            {[
              { href: "/", label: "Accueil" },
              { href: "/menu", label: "Menu" },
              { href: "/cuisine", label: "Cuisine" },
              { href: "/about", label: "À propos" },
              { href: "/login", label: "Connexion" },
            ].map((link) => (
              <a
                key={link.href}
                href={link.href}
                className="rounded-lg px-3 py-2.5 text-sm font-medium text-gray-700 hover:bg-gray-50 dark:text-gray-300 dark:hover:bg-gray-800"
              >
                {link.label}
              </a>
            ))}
          </nav>
        </div>
      )}
    </header>
  );
}
