// @daada/ui/src/components/RecipeCard.tsx
import * as React from "react";
import { cn } from "../utils/cn";
import { Badge } from "./Badge";

type Difficulty = "easy" | "medium" | "hard";
type Tier = "free" | "premium";

interface RecipeCardProps {
  id: string;
  title: string;
  imageUrl?: string;
  difficulty: Difficulty;
  prepTime: number;
  cookTime: number;
  tier: Tier;
  onClick?: (id: string) => void;
  className?: string;
}

const difficultyLabel: Record<Difficulty, string> = {
  easy: "Facile",
  medium: "Moyen",
  hard: "Difficile",
};

const difficultyColor: Record<Difficulty, string> = {
  easy: "text-green-600 dark:text-green-400",
  medium: "text-amber-600 dark:text-amber-400",
  hard: "text-red-600 dark:text-red-400",
};

export function RecipeCard({
  id,
  title,
  imageUrl,
  difficulty,
  prepTime,
  cookTime,
  tier,
  onClick,
  className,
}: RecipeCardProps) {
  const isPremium = tier === "premium";

  return (
    <div
      onClick={() => onClick?.(id)}
      className={cn(
        "group relative overflow-hidden rounded-2xl border border-gray-100 bg-white cursor-pointer",
        "transition-all duration-200 hover:shadow-lg hover:-translate-y-0.5",
        "dark:border-gray-800 dark:bg-gray-900",
        className,
      )}
    >
      {/* Image */}
      <div className="relative h-52 overflow-hidden bg-gray-100 dark:bg-gray-800">
        {imageUrl ? (
          <img
            src={imageUrl}
            alt={title}
            className={cn(
              "h-full w-full object-cover transition-transform duration-300 group-hover:scale-105",
              isPremium && "blur-sm scale-105",
            )}
            loading="lazy"
          />
        ) : (
          <div className="flex h-full items-center justify-center text-5xl">
            {isPremium ? "🔒" : "👨‍🍳"}
          </div>
        )}

        {/* Tier badge */}
        <div className="absolute right-3 top-3">
          <Badge variant={isPremium ? "premium" : "free"}>
            {isPremium ? "✨ PREMIUM" : "✓ GRATUIT"}
          </Badge>
        </div>

        {/* Premium overlay */}
        {isPremium && (
          <div className="absolute inset-0 flex flex-col items-center justify-center bg-black/30 backdrop-blur-[2px]">
            <div className="flex flex-col items-center gap-2 rounded-xl bg-white/90 px-5 py-3 dark:bg-gray-900/90">
              <span className="text-2xl">🔓</span>
              <span className="text-sm font-semibold text-gray-800 dark:text-gray-200">
                Débloquer la recette
              </span>
            </div>
          </div>
        )}
      </div>

      {/* Content */}
      <div className="p-4">
        <h3 className="font-semibold text-gray-900 dark:text-white line-clamp-2">{title}</h3>
        <div className="mt-2 flex items-center gap-3 text-xs text-gray-500 dark:text-gray-400">
          <span className={cn("font-medium", difficultyColor[difficulty])}>
            {difficultyLabel[difficulty]}
          </span>
          <span>•</span>
          <span className="flex items-center gap-1">
            <svg className="h-3.5 w-3.5" fill="none" stroke="currentColor" viewBox="0 0 24 24">
              <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z" />
            </svg>
            {prepTime + cookTime} min
          </span>
        </div>
      </div>
    </div>
  );
}
