// @daada/ui/src/components/Toast.tsx
"use client";

import * as React from "react";
import { cn } from "../utils/cn";

type ToastType = "success" | "error" | "warning" | "info";

interface ToastMessage {
  id: string;
  type: ToastType;
  title: string;
  description?: string;
}

interface ToastContextValue {
  toast: (opts: Omit<ToastMessage, "id">) => void;
  success: (title: string, description?: string) => void;
  error: (title: string, description?: string) => void;
}

const ToastContext = React.createContext<ToastContextValue | null>(null);

const typeStyles: Record<ToastType, string> = {
  success: "border-green-200 bg-green-50 dark:border-green-800 dark:bg-green-950",
  error: "border-red-200 bg-red-50 dark:border-red-800 dark:bg-red-950",
  warning: "border-amber-200 bg-amber-50 dark:border-amber-800 dark:bg-amber-950",
  info: "border-blue-200 bg-blue-50 dark:border-blue-800 dark:bg-blue-950",
};

const iconMap: Record<ToastType, string> = {
  success: "✅",
  error: "❌",
  warning: "⚠️",
  info: "ℹ️",
};

export function Toast({ message, onClose }: { message: ToastMessage; onClose: () => void }) {
  React.useEffect(() => {
    const timer = setTimeout(onClose, 4000);
    return () => clearTimeout(timer);
  }, [onClose]);

  return (
    <div
      role="alert"
      className={cn(
        "flex items-start gap-3 rounded-xl border p-4 shadow-lg animate-slide-up",
        "min-w-[300px] max-w-[400px]",
        typeStyles[message.type],
      )}
    >
      <span className="text-lg leading-none mt-0.5">{iconMap[message.type]}</span>
      <div className="flex-1">
        <p className="text-sm font-semibold text-gray-900 dark:text-white">{message.title}</p>
        {message.description && (
          <p className="mt-0.5 text-xs text-gray-600 dark:text-gray-400">{message.description}</p>
        )}
      </div>
      <button
        onClick={onClose}
        className="shrink-0 text-gray-400 hover:text-gray-600 dark:hover:text-gray-300 transition-colors"
        aria-label="Fermer"
      >
        <svg className="h-4 w-4" fill="none" stroke="currentColor" viewBox="0 0 24 24">
          <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M6 18L18 6M6 6l12 12" />
        </svg>
      </button>
    </div>
  );
}

export function ToastProvider({ children }: { children: React.ReactNode }) {
  const [messages, setMessages] = React.useState<ToastMessage[]>([]);

  const remove = React.useCallback((id: string) => {
    setMessages((prev) => prev.filter((m) => m.id !== id));
  }, []);

  const addToast = React.useCallback((opts: Omit<ToastMessage, "id">) => {
    const id = Math.random().toString(36).slice(2);
    setMessages((prev) => [...prev, { ...opts, id }]);
  }, []);

  const value: ToastContextValue = React.useMemo(
    () => ({
      toast: addToast,
      success: (title, description) => addToast({ type: "success", title, description }),
      error: (title, description) => addToast({ type: "error", title, description }),
    }),
    [addToast],
  );

  return (
    <ToastContext.Provider value={value}>
      {children}
      <div
        aria-live="polite"
        className="fixed bottom-4 right-4 z-[9999] flex flex-col gap-2"
      >
        {messages.map((msg) => (
          <Toast key={msg.id} message={msg} onClose={() => remove(msg.id)} />
        ))}
      </div>
    </ToastContext.Provider>
  );
}

export function useToast(): ToastContextValue {
  const ctx = React.useContext(ToastContext);
  if (!ctx) throw new Error("useToast must be used inside ToastProvider");
  return ctx;
}
