// @daada/shared/src/types/index.ts
// ── Core entity types ─────────────────────────────────────────────────────────

export type ID = string;

export type Locale = "fr" | "en";

export type Currency = "XAF"; // CFA Franc

export interface Timestamps {
  createdAt: string;
  updatedAt: string;
}

// ── User ──────────────────────────────────────────────────────────────────────

export type UserRole = "customer" | "restaurant_owner" | "driver" | "admin";

export interface User extends Timestamps {
  id: ID;
  phone: string;
  email?: string;
  name: string;
  avatarUrl?: string;
  role: UserRole;
  isActive: boolean;
  locale: Locale;
}

// ── Restaurant ────────────────────────────────────────────────────────────────

export interface Restaurant extends Timestamps {
  id: ID;
  name: string;
  description: string;
  logoUrl?: string;
  coverUrl?: string;
  address: string;
  phone: string;
  latitude: number;
  longitude: number;
  isOpen: boolean;
  openingHours: OpeningHours[];
  deliveryRadius: number; // km
  minOrderAmount: number; // XAF
  avgDeliveryTime: number; // minutes
  rating: number;
  reviewCount: number;
  ownerId: ID;
}

export interface OpeningHours {
  day: 0 | 1 | 2 | 3 | 4 | 5 | 6; // 0 = Sunday
  openTime: string; // "08:00"
  closeTime: string; // "22:00"
}

// ── Menu ──────────────────────────────────────────────────────────────────────

export type MenuCategory =
  | "plats"
  | "grillades"
  | "soupes"
  | "accompagnements"
  | "boissons"
  | "desserts";

export interface MenuItem extends Timestamps {
  id: ID;
  restaurantId: ID;
  name: string;
  nameEn?: string;
  description: string;
  price: number; // XAF
  imageUrl?: string;
  category: MenuCategory;
  isAvailable: boolean;
  isPopular: boolean;
  allergens?: string[];
  preparationTime: number; // minutes
}

// ── Order ─────────────────────────────────────────────────────────────────────

export type OrderStatus =
  | "pending"
  | "confirmed"
  | "preparing"
  | "ready"
  | "picked_up"
  | "delivered"
  | "cancelled";

export interface OrderItem {
  menuItemId: ID;
  name: string;
  price: number;
  quantity: number;
  notes?: string;
}

export interface Order extends Timestamps {
  id: ID;
  customerId: ID;
  restaurantId: ID;
  driverId?: ID;
  items: OrderItem[];
  status: OrderStatus;
  subtotal: number;
  deliveryFee: number;
  total: number;
  paymentMethod: PaymentMethod;
  paymentStatus: "pending" | "paid" | "failed" | "refunded";
  deliveryAddress: string;
  deliveryLatitude: number;
  deliveryLongitude: number;
  estimatedDelivery?: string;
  notes?: string;
}

// ── Payment ───────────────────────────────────────────────────────────────────

export type PaymentMethod =
  | "orange_money"
  | "mtn_momo"
  | "cash_on_delivery";

export interface Payment extends Timestamps {
  id: ID;
  orderId: ID;
  amount: number;
  currency: Currency;
  method: PaymentMethod;
  status: "pending" | "success" | "failed";
  transactionRef?: string;
  providerRef?: string;
}

// ── Recipe ────────────────────────────────────────────────────────────────────

export type RecipeDifficulty = "easy" | "medium" | "hard";
export type RecipeTier = "free" | "premium";

export interface Recipe extends Timestamps {
  id: ID;
  title: string;
  description: string;
  imageUrl?: string;
  videoUrl?: string;
  difficulty: RecipeDifficulty;
  prepTime: number; // minutes
  cookTime: number; // minutes
  servings: number;
  tier: RecipeTier;
  ingredients: Ingredient[];
  steps: string[];
  tags: string[];
  authorId: ID;
}

export interface Ingredient {
  name: string;
  quantity: string;
  unit?: string;
  isLocallyAvailable: boolean;
}

// ── Driver ────────────────────────────────────────────────────────────────────

export interface Driver extends Timestamps {
  id: ID;
  userId: ID;
  vehicleType: "moto" | "bicycle" | "car";
  licensePlate?: string;
  isOnline: boolean;
  currentLatitude?: number;
  currentLongitude?: number;
  rating: number;
  totalDeliveries: number;
}

// ── API ───────────────────────────────────────────────────────────────────────

export interface ApiResponse<T> {
  data: T;
  message?: string;
}

export interface ApiError {
  code: string;
  message: string;
  details?: Record<string, unknown>;
}

export interface PaginatedResponse<T> {
  data: T[];
  total: number;
  page: number;
  pageSize: number;
  hasMore: boolean;
}
